lib/Reporters/MutationTestingElementsReporter.cpp

```diff
@@ -36,30 +36,17 @@ json11::Json createFiles(const Result &result,
   // on their source file.
   // Step 1: we create a map: "file => all of its mutation points".
   std::map<std::string, std::vector<MutationPoint *>> mutationPointsPerFile;
-  for (auto &mutationResult : result.getMutationResults()) {
-    MutationPoint *mutant = mutationResult->getMutationPoint();
-
+  for (auto mutant : result.getMutationPoints()) {
     auto &sourceLocation = mutant->getSourceLocation();
     auto sourceCodeLine = sourceManager.getLine(sourceLocation);
     assert(sourceLocation.column < sourceCodeLine.size());
 
-    if (filesJSON.count(sourceLocation.filePath) == 0) {
-      std::pair<std::string, std::vector<MutationPoint *>> pair =
-          std::make_pair(sourceLocation.filePath,
-                         std::vector<MutationPoint *>());
-
-      mutationPointsPerFile.insert(pair);
-    }
-
-    std::vector<MutationPoint *> &currentFileMutationPoints =
-        mutationPointsPerFile.at(sourceLocation.filePath);
-    currentFileMutationPoints.push_back(mutant);
+    mutationPointsPerFile[sourceLocation.filePath].push_back(mutant);
```

Comment:
There is no need to insert empty values into the `map`: `std` will instantiate an empty vector for you if it cannot find the corresponding entry.

Reply:
👍